### PR TITLE
Change the spoken language in a model

### PR DESCRIPTION
**Pull request by amolenaar**
### PR Type
What kind of change does this PR introduce?

- [ ] Bug fix
- [x] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

The diagram language s the same as the system language. This can be a problem if the model is supposed to be made in a particular language.

Issue Number: #2162 

### What is the new behavior?

Add a setting for the language used in the model. This can be different from the system language. The language setting is saved along with the model.

The translated text is taken from `Diagram.gettext()`.

### Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### Other information

Question: should we set the default language to English, as @peter88213 suggests in #2162, or should it stick to the system default language? For me personally, having the diagrams in English would be my default.

image

### CODE REVIEW

File: gaphor/ui/elementeditor.py [hunk 1]
```diff
@@ -34,6 +35,32 @@
...
+    ("it", gettext("Italian")),
+    ("ja", gettext("Japanese")),
+    ("ko", gettext("Korean")),
+    ("pl", gettext("Polish")),
+    ("pt_BR", gettext("Portuguese (Brazil)")),
+    ("ru", gettext("Russian")),
+    ("sl", gettext("Slovenian")),
+    ("es", gettext("Spanish")),
+    ("sv", gettext("Swedish")),
+    ("tr", gettext("Turkish")),
```
Comment (amolenaar) [inline gaphor/ui/elementeditor.py:60]: We could also get those language names from the Babel database. That would safe us some translation work.

Only, how do we know what the actual language is?

Comment (danyeaw) [reply gaphor/ui/elementeditor.py:60]: You mean what the system language is set to?

I like the idea to get the names from Babel. Could we parse the name of each folder in gaphor/locale to get the list of supported languages? Then `babel.Locale.get_language_name()` should be able to translate the name, I think.

**Comment by amolenaar** [comment]
The merge conflict resolver already proved useful in this PR :)

**Comment by amolenaar** [comment]
I think the natural/spoken language is working fine. I changed the app, so it's using English by default. I think it makes sense, since English is *the* language of Open Source software and most software used internationally is written in English. The language can be changed and the language preferences will be saved as part of the model.

**Review by danyeaw — APPROVED** [review]
This is a great feature @amolenaar!

File: gaphor/ui/elementeditor.py [hunk 2]
```diff
@@ -295,7 +298,20 @@ def tx_update_style_sheet(style_sheet, text):
         self._style_sheet_update = DelayedFunction(800, tx_update_style_sheet)
         self._in_update = 0
 
+        self._languages = [
+            (None, "English"),
+            *sorted(
+                (
+                    (d.name, Locale.parse(d.name).get_language_name())
```
[the remark this answers is not in the document]
Comment (amolenaar) [reply gaphor/ui/elementeditor.py]: Done. Looks better indeed.

**Pull request merged**